json-lint/schemas/inventory/adservers/spotx.json

```diff
@@ -0,0 +1,61 @@
+{
+    "$id": "#spotx",
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "title": "Spotx adserver",
+    "description": "https://github.com/ampproject/amphtml/blob/master/ads/spotx.md",
+    "additionalProperties": false,
+    "required": [
+      "type",
+      "width",
+      "heigth",
+      "spotx_channel_id"
+    ],
+    "defaultSnippets": [
+      {
+        "label": "Spotx template",
+        "description": "contains required keys",
+        "body": {
+          "type": "spotx",
+          "width": "$1",
+          "height": "$2",
+          "spotx_channel_id": "$3"
+        }
+      }
+    ],
+    "properties": {
+      "type": {
+        "const": "spotx"
+      },
+      "width": {
+        "type": ["integer", "string"],
+        "pattern": "^(\\$\\{.+\\})$",
+        "title": "integer",
+        "description": "example: 300"
+      },
+      "height": {
+        "type": ["integer", "string"],
+        "pattern": "^(\\$\\{.+\\})$",
+        "title": "integer",
+        "description": "example: 250"
+      },
+      "spotx_channel_id": {
+        "type": ["integer", "string"]
+      },
+      "spotx_ad_format": {
+        "type": "string"
+      },
+      "spotx_content_width": {
+        "type": "string"
+      },
+      "spotx_content_height": {
+        "type": "string"
+      },
+      "spotx_unmute_on_mouse": {
```

Comment:
I don't know if you saw this working... I guess java is transforming spotx_unmute_on_mouse to data-spotx-unmute-on-mouse, because there was a problem with PHP and press, because PHP couldn't read the - and so they read _ and transform it later to -...
I'm not saying this schema is wrong, just saying to take a look at the JAVA part and maybe deprecate the transformation of _ to - , since we don't use anymore MarfeelPress 1

Reply:
data-spotx-unmute-on-mouse would be a wrong output
data-spotx_unmute_on_mouse 👍

Reply:
yeah I checked all of that already, Gutenberg is taking care of it